### PR TITLE
clean up sdsenabled flag

### PR DESCRIPTION
**Pull request by ramaraochavali**
This PR cleans up the SDSEnabled flag and relevant code.

Only thing remaining is  in `simple_test.go` few validations (envoy metrics) are failing because it expects sds-grpc cluster to be available for some clusters/listeners. But it does not affect the basic verifications. So  left a TODO in that file and it will be easier once Pilot SDS implements. We can reenable those assertions. I tried to use nodeagent test env to start SDS but that is looking cumbersome - since any we plan to implement SDS in pilot it becomes lot easier.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 1]
```diff
@@ -1082,49 +1061,29 @@ func buildUpstreamClusterTLSContext(opts *buildClusterOpts, tls *networking.Clie
...
-				CertificatePath:   model.GetOrDefault(proxy.Metadata.TLSClientCertChain, tls.ClientCertificate),
-				PrivateKeyPath:    model.GetOrDefault(proxy.Metadata.TLSClientKey, tls.PrivateKey),
-				CaCertificatePath: model.GetOrDefault(proxy.Metadata.TLSClientRootCert, tls.CaCertificates),
-			}
-			tlsContext.CommonTlsContext.TlsCertificateSdsSecretConfigs = append(tlsContext.CommonTlsContext.TlsCertificateSdsSecretConfigs,
-				authn_model.ConstructSdsSecretConfig(res.GetResourceName()))
+		// These are certs being mounted from within the pod. Rather than reading directly in Envoy,
+		// which does not support rotation, we will serve them over SDS by reading the files.
+		res := model.SdsCertificateConfig{
+			CertificatePath:   model.GetOrDefault(proxy.Metadata.TLSClientCertChain, tls.ClientCertificate),
```
Comment (rshriram) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:1067]: remember per our convo, we dont need this get or default.. its just the tls client cert

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/cluster.go:1067]: Yes. Will do. First I want to cleanup the code that is not used. I will do that getOrDefault stuff in separate PR

**Comment by ramaraochavali** [comment]
@istio/wg-security-maintainers PTAL when you get chance

**Comment by ramaraochavali** [comment]
@istio/wg-security-maintainers gentle ping - Need to do some follow-ups once this is merged. Could you PTAL?

**Comment by ramaraochavali** [comment]
@istio/wg-security-maintainers PTAL - this is getting in to frequent conflicts because of changes involved.

**Pull request merged**